tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -66,6 +69,11 @@ final class ScalaJSClassEmitter private (
     linkingUnit.classDefs.map(c => c.encodedName -> c).toMap
   }
 
+  private[javascript] def isInterface(className: String): Boolean = {
+    linkingUnit != null &&
+    linkedClassByName(className).kind == ClassKind.Interface
```

Comment:
I would go with solution 1 +  clearly document the assumptions the emitter makes and how they are satisfied. Well might need a more principled way of addressing this in the future anyways.